### PR TITLE
Cleaned Up JavaScript Testing Notes

### PR DESCRIPTION
**Pull request by clintonb**
- Use single make command to install prerequisites and run tests/linting
- Updated gulpfile to watch all JS files
- Updated karma-coverage to output summary to stdout

@rocha @dsjen

### CODE REVIEW

**Comment by dsjen** [comment]
It would be good to include @benpatterson too.

File: README.md [hunk 1]
```diff
@@ -56,12 +56,19 @@ Please read [How To Contribute](https://github.com/edx/edx-platform/blob/master/
...
 This project uses `nose` to find and run tests. `pep8` and `pylint` are used to verify code quality. All three can be
 run with the command below:
 
     $ make validate
 
+
+### JavaScript Tests & Code Quality
+JavaScript tests and linting can be run with the following command:
+
+        $ make validate_js
```
Comment (rocha) [inline README.md:81]: One suggestion is to have a `validate_python` and a `validate_js` target, that get called by the `validate` target. That way CI can run all the test with just one command, and we get to run tests just for what we need.

Comment (clintonb) [inline README.md:81]: Yup. I was thinking the same thing this morning.

**Comment by rocha** [comment]
Looks good, let me know if my suggestion make sense. Check with @benpatterson before merging.

**Comment by benpatterson** [comment]
Looks good to me. I would just point out where to find the resulting quality and coverage reports in the README.

File: README.md [hunk 2]
```diff
@@ -57,10 +57,20 @@ Testing
 -------
 
 ### Unit Tests & Code Quality
-This project uses `nose` to find and run tests. `pep8` and `pylint` are used to verify code quality. All three can be
-run with the command below:
+The complete unit test and quality suite can be run with:
+
+        $ make validate
+
```
Comment (benpatterson) [inline README.md:72]: I would point out where to find the various reports.

Comment (clintonb) [inline README.md:72]: Done.

**Comment by clintonb** [comment]
@benpatterson Can you update the Jenkins job to use `make validate` instead of `make test` (which no longer exists)?

**Comment by benpatterson** [comment]
@clintonb I am looking at Jenkins now and see you already took care of it. Solid.

**Pull request merged**